MainWindow.cpp

```diff
@@ -196,12 +261,19 @@ void MainWindow::onForceMdiChildDestroyed(void)
         }
     }
 
-    foreach (ForceEffect *force, m_forces)
+    QMutableListIterator<QPointer<ForceEffect>> i(m_forces);
+    while (i.hasNext())
     {
-        if (!forcesInUse.contains(force))
+        QPointer<ForceEffect> val = i.next();
+        if (!forcesInUse.contains(val))
         {
-            force->deleteLater();
+            if (m_gameController)
+            {
+                m_gameController->removeForce(val);            
+            }
+            i.remove();
         }
+    
     }
```

Comment:
why complicated qmutablelistinterator? the 'deleteLater' makes sure the list is valid while we iterate, doesn't it?